heron/stmgr/src/cpp/manager/stmgr-server.cpp

```diff
@@ -667,5 +682,117 @@ void StMgrServer::AttemptStopBackPressureFromSpouts() {
   }
 }
 
+void StMgrServer::InitiateStatefulCheckpoint(const sp_string& _checkpoint_tag) {
+  for (auto iter = instance_info_.begin(); iter != instance_info_.end(); ++iter) {
+    if (iter->second->is_local_spout() && iter->second->conn_) {
```

Comment:
A stmgr services a bunch of tasks, some spouts and others bolts. local spouts are the former. This check is basically checking if this task is a local spout and only send the initiate checkpoint message to them. I will add a comment here.